src/v2/guide/instance.md

```diff
@@ -16,46 +16,68 @@ var vm = new Vue({
 
-The `Vue` constructor can be extended to create reusable **component constructors** with pre-defined options:
+A Vue application consists of a **root Vue instance** created with `new Vue`, optionally organized into a tree of nested, reusable components. For example, a todo app's component tree might look like this:
 
-``` js
-var MyComponent = Vue.extend({
-  // extension options
-})
-
-// all instances of `MyComponent` are created with
-// the pre-defined extension options
-var myComponentInstance = new MyComponent()
+```
+Root Instance
+|- TodoList
+   |- TodoItem
+      |- DeleteTodoButton
+      |- EditTodoButton
+   |- TodoListFooter
+      |- ClearTodosButton
+      |- TodoListStatistics
 ```
 
-Although it is possible to create extended instances imperatively, most of the time it is recommended to compose them declaratively in templates as custom elements. We will talk about [the component system](components.html) in detail later. For now, you just need to know that all Vue components are essentially extended Vue instances.
+We'll talk about [the component system](components.html) in detail later. For now, just know know that all Vue components are also Vue instances, and so accept the same options object (except for a few root-specific options).
 
-## Properties and Methods
+## Data and Methods
 
-Each Vue instance **proxies** all the properties found in its `data` object:
+When a Vue instance is created, it adds all the properties found in its `data` object to Vue's **reactivity system**. When the values of those properties change, the view will "react", updating to match the new values.
 
 ``` js
+// Our data object
 var data = { a: 1 }
+
+// The object is added to a Vue instance
 var vm = new Vue({
   data: data
 })
 
-vm.a === data.a // -> true
+// These reference the same object!
+vm.a === data.a // => true
 
-// setting the property also affects original data
+// Setting the property on the instance
+// also affects the original data
 vm.a = 2
-data.a // -> 2
+data.a // => 2
 
 // ... and vice-versa
 data.a = 3
-vm.a // -> 3
+vm.a // => 3
+```
+
+When this data changes, the view will re-render. It should be noted that properties in `data` are only **reactive** if they existed when the instance was created. That means if you add a new property, like:
+
+``` js
+vm.b = 'hi'
 ```
 
-It should be noted that only these proxied properties are **reactive**, meaning any changes to their values will trigger the view to re-render. If you attach a new property to the instance after it has been created, it will not trigger any view updates. We will discuss the reactivity system in detail later.
+Then changes to `b` will not trigger any view updates. If you know you'll need a property later, but it starts out empty or non-existent, you'll just need to set some initial value. For example:
+
+``` js
+data: {
+  newTodoText: '',
+  visitCount: 0,
+  hideCompletedTodos: false,
+  todos: [],
+  errorMessage: null
```

Comment:
I'd suggest `error` instead of `errorMessage`, as a "message" would imply a string type.